vmware-event-router/internal/processor/aws/aws_event_bridge.go

```diff
@@ -126,16 +122,38 @@ func NewEventBridgeProcessor(ctx context.Context, cfg *config.ProcessorConfigEve
 		return nil, errors.New("event bus must be specified")
 	}
 
-	awsSession, err := session.NewSession(&aws.Config{
-		Region: aws.String(cfg.Region),
-		Credentials: credentials.NewStaticCredentials(
-			accessKey,
-			secretKey,
-			"", // a token will be created when the session is used.
-		),
-	})
-	if err != nil {
-		return nil, errors.Wrap(err, "create AWS session")
+	// Check the Auth Method to determine how the Session should be established
+	if cfg.Auth.Type == "aws_access_key" {
+		if cfg.Auth == nil || cfg.Auth.AWSAccessKeyAuth == nil {
+			return nil, fmt.Errorf("invalid %s credentials: accessKey and secretKey must be set", config.AWSAccessKeyAuth)
+		}
+		accessKey := cfg.Auth.AWSAccessKeyAuth.AccessKey
+		secretKey := cfg.Auth.AWSAccessKeyAuth.SecretKey
+
+		awsSessionAccessKey, err := session.NewSession(&aws.Config{
+			Region: aws.String(cfg.Region),
+			Credentials: credentials.NewStaticCredentials(
+				accessKey,
+				secretKey,
+				"", // a token will be created when the session is used.
+			),
+		})
+		if err != nil {
+			return nil, errors.Wrap(err, "create AWS session")
+		}
+		// Set the AWS Session to the IAM Role authenticated session
+		awsSession = awsSessionAccessKey
+	}
+	if cfg.Auth.Type == "aws_iam_role" {
```

Comment:
see above

Reply:
Adjusted to assert on type in 26e4558e0d2e864e25d12f04e3c2e8ae62fe351d